### PR TITLE
Update labels to use kubernetes recommended labels

### PR DESCRIPTION
**Pull request by therealdwright**
## What does this PR do?

Kubernetes have not used the `app` or `k8s-app` for some time
and have provided guidance on what the recommended labels are
for some time as seen in their [docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/).

This commit updates all manifests/docs to use this approach.
## Why is it important?

Users who choose to adopt kubernetes label best practises will not have to fork elastic manifests and can instead rely on the upstream ones directly.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [X] I have added an integration test or an E2E test

## Disruptive User Impact

Users will need to upgrade using `kubectl replace` as some of the labels are immutable

### CODE REVIEW

**Review by gizas — APPROVED** [review]
Looks good!

**Review by blakerouse — CHANGES_REQUESTED** [review]
Definitely need a changelog entry for this change, as this affects users.

**Comment by pierrehilbert** [comment]
@therealdwright Thanks for your contribution and signing the CLA.
As @blakerouse mentioned, could you please add de changelog entry to let our users be aware of this change?

**Comment by therealdwright** [comment]
> could you please add de changelog entry to let our users be aware of this change?

I've pushed a change log entry as requested.

**Review by blakerouse — APPROVED** [review]
Thanks for the changelog.

**Comment by therealdwright** [comment]
@blakerouse I have rebased on main as the `SonarQube Code Analysis` and `buildkite/elastic-agent ` were stuck in waiting for status to be reported. LMK if this is something maintainers must manually trigger.

**Comment by blakerouse** [comment]
@therealdwright Yes it requires an authorized user to approve and run the workflows. This ensures we can review the change before we allow it to run on an actual machine.

**Comment by ycombinator** [comment]
buildkite test this

**Comment by ycombinator** [comment]
buildkite test this

**Comment by ycombinator** [comment]
CI failures look relevant: https://buildkite.com/elastic/elastic-agent/builds/9028#018f97b0-430e-4777-9267-e618e45b3341/512

**Comment by therealdwright** [comment]
> CI failures look relevant

How do contributors get access to CI logs?

**Comment by ycombinator** [comment]
> > CI failures look relevant
> 
> How do contributors get access to CI logs?

Oops, sorry about that, @therealdwright!  Here are the logs that are relevant:

```
diff --git a/deploy/kubernetes/elastic-agent-managed-kubernetes.yaml b/deploy/kubernetes/elastic-agent-managed-kubernetes.yaml
--
  | index 074a15daa1..b54834c313 100644
  | a/deploy/kubernetes/elastic-agent-managed-kubernetes.yaml | 0s
  | b/deploy/kubernetes/elastic-agent-managed-kubernetes.yaml | 0s
  | @@ -212,7 +212,7 @@ kind: ClusterRole
  | metadata:
  | name: elastic-agent
  | labels:
  | -    app.kubernetes.io/name.kubernetes.io/name: elastic-agent
  | +    app.kubernetes.io/name: elastic-agent
  | rules:
  | - apiGroups: [""]
  | resources:
  | @@ -284,7 +284,7 @@ metadata:
  | # Should be the namespace where elastic-agent is running
  | namespace: kube-system
  | labels:
  | -    app.kubernetes.io/name.kubernetes.io/name: elastic-agent
  | +    app.kubernetes.io/name: elastic-agent
  | rules:
  | - apiGroups:
  | - coordination.k8s.io
  | @@ -298,7 +298,7 @@ metadata:
  | name: elastic-agent-kubeadm-config
  | namespace: kube-system
  | labels:
  | -    app.kubernetes.io/name.kubernetes.io/name: elastic-agent
  | +    app.kubernetes.io/name: elastic-agent
  | rules:
  | - apiGroups: [""]
  | resources:
  | @@ -313,5 +313,5 @@ metadata:
  | name: elastic-agent
  | namespace: kube-system
  | labels:
  | -    app.kubernetes.io/name.kubernetes.io/name: elastic-agent
  | +    app.kubernetes.io/name: elastic-agent
  | ---
  | deploy/kubernetes/elastic-agent-managed-kubernetes.yaml: needs update
  | make[1]: *** [Makefile:59: check-no-changes] Error 1
  | make[1]: Leaving directory '/opt/buildkite-agent/builds/bk-agent-prod-gcp-1716236865505762234/elastic/elastic-agent'
  | make: *** [Makefile:40: check-ci] Error 2
```

To fix that, I ran `make generate-k8s` from the `deploy/kubernetes` folder and pushed up 04de6e6ee127c749a16e11bf5f1705deafa5af5e with the changes.

**Comment by jlind23** [comment]
buildkite test it

**Comment by jlind23** [comment]
@gizas Unfortunately this PR hasn't been looked at for quite a while, would you mind reviewing it again in light of all the recent changes that could have affected the kustomize templates?

File: deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/extra/elastic-agent-standalone-statefulset.yaml [hunk 1]
```diff
@@ -66,6 +66,8 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            - name: STATE_PATH
```
Comment (pkoutsovasilis) [inline deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/extra/elastic-agent-standalone-statefulset.yaml:69]: @gizas this isn't a label change, so is this change required?

File: deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml [hunk 2]
```diff
@@ -66,6 +66,8 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            - name: STATE_PATH
```
Comment (pkoutsovasilis) [inline deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml:69]: @gizas this isn't a label change, so is this change required?

Comment (gizas) [reply deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml:69]: Indeed is not needed

File: deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/extra/elastic-agent-standalone-statefulset.yaml (hunk 1 above)
Comment (gizas) [reply deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/extra/elastic-agent-standalone-statefulset.yaml:69]: Same as above, not needed

**Review by swiatekm — COMMENTED** [review]
Looks good to me, minus @pkoutsovasilis comment about the STATE_PATH variable.

File: deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml (hunk 2 above)
Comment (therealdwright) [reply deploy/kubernetes/elastic-agent-kustomize/ksm-autosharding/elastic-agent-standalone/base/elastic-agent-standalone-daemonset.yaml:69]: It was not me that added this but I will rebase and remove.

**Comment by jlind23** [comment]
Sonar and CI check will never run as there was no code change hence force merging this as both @swiatekm and @blakerouse  approved.

**Comment by jlind23** [comment]
Thanks @therealdwright for pushing this through!
